src/Mod/Sketcher/App/planegcs/Constraints.cpp

```diff
@@ -129,6 +129,171 @@ double ConstraintEqual::grad(double *param)
     return scale * deriv;
 }
 
+// Weighted Linear Combination
+
+const std::vector< std::vector<double> > ConstraintWeightedLinearCombination::defaultfactors =
+{{},
+ {1.0},
+ {0.5, 0.5},
+ {0.16666666666666666, 0.6666666666666667, 0.16666666666666669},
+ {0.041666666666666664, 0.45833333333333337, 0.4583333333333333, 0.04166666666666667},
+ {0.008333333333333333, 0.21666666666666667, 0.55, 0.2166666666666667, 0.008333333333333333},
+ {0.0013888888888888887, 0.07916666666666666, 0.41944444444444445, 0.41944444444444445, 0.07916666666666669, 0.0013888888888888885},
+ {0.0001984126984126984, 0.023809523809523815, 0.2363095238095238, 0.4793650793650793, 0.2363095238095238, 0.02380952380952381, 0.0001984126984126984},
+ {2.48015873015873e-05, 0.006125992063492064, 0.10647321428571428, 0.3873759920634921, 0.38737599206349216, 0.10647321428571431, 0.006125992063492064, 2.48015873015873e-05},
+ {2.7557319223985893e-06, 0.0013833774250440916, 0.04025573192239859, 0.24314925044091712, 0.43041776895943573, 0.24314925044091715, 0.04025573192239859, 0.0013833774250440916, 2.7557319223985905e-06}};
+
+
+ConstraintWeightedLinearCombination::ConstraintWeightedLinearCombination(size_t givennumpoints, const std::vector<double *>& givenpvec)
+    : numpoints(givennumpoints)
+{
+    pvec = givenpvec;
+    setupInputs();
+}
+
+ConstraintWeightedLinearCombination::ConstraintWeightedLinearCombination(size_t givennumpoints, const std::vector<double *>& givenpvec, const std::vector<double>& givenfactors)
+    : factors(givenfactors)
+    , numpoints(givennumpoints)
+{
+    pvec = givenpvec;
+    setupInputs();
+}
+
+void ConstraintWeightedLinearCombination::setupInputs()
+{
+    assert(pvec.size() == 2*numpoints + 1);
+    if (factors.empty())
+        factors = defaultfactors[numpoints];
+    assert(factors.size() == numpoints);
+    origpvec = pvec;
+    rescale();
+}
+
+ConstraintType ConstraintWeightedLinearCombination::getTypeId()
+{
+    return WeightedLinearCombination;
+}
+
+void ConstraintWeightedLinearCombination::rescale(double coef)
+{
+    scale = coef * 1.;
+}
+
+double ConstraintWeightedLinearCombination::error()
+{
+    double offset = 1;
+    double woffset = 1 + numpoints;
+
+    double sum = 0;
+    double wsum = 0;
+
+    for (size_t i = 0; i < numpoints; ++i) {
+        double wcontrib = *pvec[woffset + i] * factors[i];
+        wsum += wcontrib;
+        sum += *pvec[offset + i] * wcontrib;
+    }
+
+    return scale * ((*pvec[0]) * wsum - sum);
+}
+
+double ConstraintWeightedLinearCombination::grad(double *param)
+{
+    double offset = 1;
+    double woffset = 1 + numpoints;
+
+    // TODO: Do we assume pvec are different so param can only be one?
```

Comment:
My concern was: can it happen that the same pointer occurs more than once in the pvec `[param0, param1,..., paramN=param0,...]`. If not, then there is no use checking further if, say `param==poleat(i)` is detected, and we can return value immediately.

Reply:
This should never be the case. Repetition of one parameter in pvec must be wrong. I cannot find any legit reason why we should. 

Gradients of each constraint are called for all the parameters of the system (including remote parameters that have a zero partial derivative, which in fact is the biggest contribution to the sparseness of the system). 

Basically we need to "find" if the param we are passed is one of "ours", and return the right partial derivative depending on which one it is. If there is a small amount (as in other constraints), all combinations may be compared, if not a loop or any suitable algorithm may be used instead.

Reply:
> This should never be the case. Repetition of one parameter in pvec must be wrong. I cannot find any legit reason why we should.

There are however other constraints where this happens. See for example `ConstraintEqual`.

Reply:
OK I'm going with your suggested route here, since at least for B-splines params will never repeat. For other constraints we should check case by case.

Reply:
I am ok with the outcome in your last comment.

However, could you point out where a parameter is repeated in pvec of ConstraintEqual? I say this because if confirmed, it may be a bug. The pvec of ConstraintEqual should have two different parameters. Another different thing is that the treatment of equal constraints is different when partitioning into subsystems.